content/en/docs/usage/kraft_advanced.md

```diff
@@ -0,0 +1,732 @@
+make: Entering directory '/home/stefan/.unikraft/unikraft'
+#
+# configuration written to /media/stefan/projects/unikraft/my-helloworld-app/.config
+#
+make: Leaving directory '/home/stefan/.unikraft/unikraft'
+
+$ diff .config .config.old
+78c78
+< CONFIG_HZ=300
+---
+> CONFIG_HZ=100
+137c137
+< CONFIG_LIBUKMMAP=y
+---
+> # CONFIG_LIBUKMMAP is not set
+```
+
+You can use a specific directory of your target application (instead of the default current working directory) by using the `-w, --workdir` option:
+
+```console
+$ kraft configure -p kvm -m x86_64 -w /home/stefan/projects/unikraft/hello/
+make: Entering directory '/home/stefan/.unikraft/unikraft'
+[...]
+#
+# configuration written to /home/stefan/projects/unikraft/hello/.config
+#
+make: Leaving directory '/home/stefan/.unikraft/unikraft'
+```
+
+You can force the overwriting of the existing configuration by using `-F, --force`.
+**Note:** If you are changing the target architecture, make sure you run `kraft clean` before rebuilding.
+
+```console
+$ kraft configure -p kvm -m x86_64 -F
+make: Entering directory '/home/stefan/.unikraft/unikraft'
+#
+# configuration written to /media/stefan/projects/unikraft/my-helloworld-app/.config
+#
+make: Leaving directory '/home/stefan/.unikraft/unikraft'
+```
+
+### Overview of the `kraft configure` Command
+
+You can get a full list of features by running `kraft configure -h` or `kraft configure --help`.
+
+```text
+Usage: kraft configure [OPTIONS]
+
+  Configure the unikernel using the KConfig options set in the kraft.yaml
+  file.  Alternatively, you can use the -k|--menuconfig flag to open the TUI
+  to manually select the configuration for this unikernel.
+
+  When the unikernel is configured, a .config file is written to the working
+  directory with the selected KConfig options.
+
+Options:
+  -p, --plat PLAT         Target platform.
+  -t, --target TARGET     Target name.
+  -m, --arch ARCH         Target architecture.
+  -F, --force             Force writing new configuration.
+  -k, --menuconfig        Use Unikraft's ncurses Kconfig editor.
+  -w, --workdir PATH      Specify an alternative directory for the application
+			  [default is cwd].
+  -y, --yes KOPTION       Specify an option to enable.
+  -n, --no KOPTION        Specify an option to disable.
+  -s, --set KOPTION       Set an option's value.
+  -u, --use-version COMP  Use the specified version for the component, e.g.
+			  -u unikraft@staging (will override kraft.yaml).
+  -h, --help              Show this message and exit.
+```
+
+## `kraft build`
+
+You can build your application by using `kraft build`.
+You can use the `--fetch / --no-fetch` and `--prepare / --no-prepare` flags to run the fetch and prepare steps before building.
+You can use the `-j, --fast` flag.
+This will pass the `-j` flag to `make`, resulting in running multiple jobs that can run simultaneously.
```

Comment:
```suggestion
This will pass the `-j` option to `make`, followed by the number of jobs that can run simultaneously.
Typically, the number of CPUS cores is passed (i.e. the output of the `nproc` command).
```

Reply:
Without `$(nproc)` it would create too many jobs, and it often blocks the system during the build.

Reply:
I know, but are you sure this works with `kraft`? I tried it and and `kraft run -j x` seems to disregard the `x` and just run as many jobs as it can.
It seems to just pass the `-j` flag to `make` in the source code too: `-j` sets the `n_proc` to -1 [here](https://github.com/unikraft/kraft/blob/staging/kraft/cmd/build.py#L64) and then `-j` is appended to the `make` command [here](https://github.com/unikraft/kraft/blob/c648d2ea1d360685502b94727bd113a13b0f82ee/kraft/app/app.py#L223) with no other arguments.

Reply:
Oh, I see. Gotcha. Leave it like that, then.